client.go

```diff
@@ -0,0 +1,103 @@
+package gotruevault
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+)
+
+const (
+	contentTypeApplicationJSON = "application/json"
+)
+
+var (
+	// ErrUnauthorized ...
+	ErrUnauthorized = errors.New("error: authorization failed")
+
+	// ErrServerError ...
+	ErrServerError = errors.New("error: server error")
+
+	// ErrBadRequest ...
+	ErrBadRequest = errors.New("error: bad request")
+)
+
+// Client is the interface that wraps http calls to TrueVault
+//go:generate mockery --name Client
+type Client interface {
+	SearchDocument(ctx context.Context, vaultID string, filter SearchFilter) (SearchDocumentResult, error)
+}
+
+type trueVaultClient struct {
+	httpClient *http.Client
+	urlBuilder URLBuilder
+	apiKey     string
+}
+
+// NewClient creates a TrueVault client
+func NewClient(h *http.Client, a string) Client {
+	return &trueVaultClient{
+		httpClient: h,
+		urlBuilder: &DefaultURLBuilder{},
+		apiKey:     base64.StdEncoding.EncodeToString([]byte(a + ":")),
+	}
+}
+
+func (c *trueVaultClient) SearchDocument(ctx context.Context, vaultID string, filter SearchFilter) (SearchDocumentResult, error) {
```

Comment:
I was thinking we would break each unit into it's own file. Thoughts?
```
user.go
document.go
blob.go
etc
```

Reply:
addressed this on this commit https://github.com/FirstVisit/go-truevault/pull/1/commits/89d16baa8bb98a8bab5d922445114d9e7db2580b